src/main/java/picard/util/SingleBarcodeDistanceMetric.java

```diff
@@ -85,7 +87,12 @@ private static boolean anySmaller(final byte[] values, final int minValue) {
         return false;
     }
 
-    private byte[] maskSmaller() {
+    static private byte[] maskIfAnySmaller(final byte [] readBases, final byte[] readQualities, final int minimumBaseQuality) {
+
+        if (!anySmaller(readQualities,minimumBaseQuality)) {
```

Comment:
If `values` is long and you're concerned about performance you could make `anySmaller` return the index where the first smaller element was found, and start the `for` loop below at that index. (And use `-1` to indicate no smaller item found.)

Reply:
I don't think that this is a huge source of compute time.